### PR TITLE
Add Tinyhttp

### PR DESCRIPTION
**Pull request by talentlessguy**
**By submitting this pull request, I promise I have read the [contribution guidelines](https://github.com/sindresorhus/awesome-nodejs/blob/master/contributing.md) twice and ensured my submission follows it. I realize not doing so wastes the maintainers' time that they could have spent making the world better. 🖖**

⬆⬆⬆⬆⬆⬆⬆⬆⬆⬆

tinyhttp is a modern Express-like web framework for Node.js, written in TypeScript and compiled to native ESM. It uses a bare minimum amount of dependencies trying to avoid legacy hell.

:earth_africa:  Website: https://tinyhttp.v1rtl.site
:octocat:  Repo: https://github.com/talentlessguy/tinyhttp

the project itself, at the moment of adding this PR, has 268 stars and 8 (including myself) contributors. Was created on June 13, so more than 30 days passed.

I used the requirements list from contributing.md:

* **If you submit a project that is similar to an existing project in the list, argue how it's better.** tinyhttp both has TypeScript support out of the box and Native ESM module available. Express has neither, Koa only ESM support but a different API. tinyhttp has a much smaller amount of dependencies than Express because it tries to use native modules / JavaScript language as much as possible. According to [benchmarks](https://github.com/talentlessguy/tinyhttp/tree/master/benchmark#results), tinyhttp is ~2.5x faster than Express.
* **Suggested packages should be tested and documented.**. There is a [0.X API](https://tinyhttp.v1rtl.site/docs) documentation, and the code coverage is 73% (I can cover it in tests more and then return back here in case it's not enough)

Let me know if I missed anything (or if I did anything wrong)!

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -331,6 +331,7 @@
 - [Nest](https://github.com/nestjs/nest) - Angular-inspired framework for building efficient and scalable server-side apps.
 - [Zeronode](https://github.com/sfast/zeronode) - Minimal building block for reliable and fault-tolerant microservices.
 - [TypeGraphQL](https://github.com/19majkel94/type-graphql) - Modern framework for creating GraphQL APIs with TypeScript, using classes and decorators.
+- [Tinyhttp](https://github.com/talentlessguy/tinyhttp) - Modern Express-like web framework compiled to native ESM and written in TypeScript that uses a bare minimum amount of dependencies trying to avoid legacy hell.
```
Comment (sindresorhus) [inline readme.md:334]: The description is too verbose. Try to focus on what matters.

Comment (talentlessguy) [reply readme.md:334]: alright, gonna reduce it a bit, thanks for feedback :3

**Comment by talentlessguy** [comment]
@sindresorhus made it a little bit shorter, so I think it's focusing on what matters

**Comment by talentlessguy** [comment]
yay thanks :D

**Pull request merged**